tests/integration/ha/test_scaling.py

```diff
@@ -0,0 +1,190 @@
+    assert num_units == init_units_count - amount
+
+    assert await access_all_dashboards(ops_test, pytest.relation.id)
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_horizontal_scale_down_to_zero_http(ops_test: OpsTest) -> None:
+    """Testing that scaling down to 0 units is possible."""
+    init_units_count = len(ops_test.model.applications[APP_NAME].units)
+    unit_id = HTTP_UNITS[0]
+
+    # scale down
+    await ops_test.model.applications[APP_NAME].destroy_unit(f"{APP_NAME}/{unit_id}")
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", timeout=1000, wait_for_exact_units=0
+    )
+    num_units = len(ops_test.model.applications[APP_NAME].units)
+    assert num_units == init_units_count - 1
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_horizontal_scale_up_https(ops_test: OpsTest) -> None:
+    """Testing that new added are functional with TLS on."""
+    await ops_test.model.applications[APP_NAME].add_unit(count=1)
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000)
+
+    # Relate it to OpenSearch to set up TLS.
+    await ops_test.model.integrate(APP_NAME, TLS_CERTIFICATES_APP_NAME)
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME, TLS_CERTIFICATES_APP_NAME], status="active", timeout=1000
+    )
+
+    init_units_count = len(ops_test.model.applications[APP_NAME].units)
+
+    amount = len(HTTPS_UNITS[1:])
+
+    # scale up
+    await ops_test.model.applications[APP_NAME].add_unit(count=amount)
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME, TLS_CERTIFICATES_APP_NAME], status="active", timeout=1000
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME],
+        status="active",
+        timeout=1000,
+        wait_for_at_least_units=len(ops_test.model.applications[APP_NAME].units),
+    )
+
+    num_units = len(ops_test.model.applications[APP_NAME].units)
+    assert num_units == init_units_count + amount
+
+    assert await access_all_dashboards(ops_test, pytest.relation.id, https=True)
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_horizontal_scale_down_https(ops_test: OpsTest) -> None:
+    """Testing that decreasing units keeps functionality with TLS on."""
+    init_units_count = len(ops_test.model.applications[APP_NAME].units)
+
+    amount = len(HTTPS_UNITS[:-1])
+
+    # scale down
+    await ops_test.model.applications[APP_NAME].destroy_unit(
+        *[f"{APP_NAME}/{cnt}" for cnt in HTTPS_UNITS[:-1]]
+    )
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000)
+    num_units = len(ops_test.model.applications[APP_NAME].units)
+    assert num_units == init_units_count - amount
+
+    assert await access_all_dashboards(ops_test, pytest.relation.id, https=True)
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_horizontal_scale_down_to_zero_https(ops_test: OpsTest) -> None:
```

Comment:
Can we have a scale up from 0 https, like we have for http?

Reply:
Done.